src/nvim/message_defs.h

```diff
@@ -10,14 +11,19 @@ typedef struct {
 } HlMessageChunk;
 
 typedef kvec_t(HlMessageChunk) HlMessage;
+typedef uint64_t MsgID;
 
+#define MSG_KIND_PROGRESS "progress"
```

Comment:
Where are UI protocols kept actually? What makes a sting valid message-kind?

Reply:
Not sure what you're asking sorry. Until now message kind strings are just set directly at the callsite, not stored/defined anywhere. One exception is the verbose kind to compare to `msg_ext_kind` directly rather than needing `strcmp()`
https://github.com/neovim/neovim/blob/435f03ee106a548c2e5fd6c10005f5ae2a29a5a4/src/nvim/message.c#L3325-L3334

Reply:
I noticed that. But I was wondering there should be at least some validation layer. Otherwise, kind can be set to anything. I guess that's the case here. The verbose case is different it lt's controlled by verbose_enter() and verbose_exit() that's why it can work with pointer comparison. It doesn't make sense for progress to be something like that.